src/librustdoc/html/render/print_item.rs

```diff
@@ -1284,10 +1284,16 @@ fn render_stability_since(
     )
 }
 
-fn compare_impl<'a, 'b>(lhs: &'a &&Impl, rhs: &'b &&Impl, cx: &Context<'_>) -> Ordering {
+fn compare_impl<'a, 'b>(lhs: &'a &Impl, rhs: &'b &Impl, cx: &Context<'_>) -> Ordering {
     let lhss = format!("{}", lhs.inner_impl().print(false, cx));
     let rhss = format!("{}", rhs.inner_impl().print(false, cx));
 
+    if lhs.inner_impl().negative_polarity && !rhs.inner_impl().negative_polarity {
+        return Ordering::Less;
+    } else if rhs.inner_impl().negative_polarity && !lhs.inner_impl().negative_polarity {
+        return Ordering::Greater;
+    }
```

Comment:
Could you change this to use a pattern match instead? I think it would be easier to understand the logic and reduce bugs. Something like this:

```rust
match (lhs.inner_impl().negative_polarity, rhs.inner_impl().negative_polarity) {
    (true, false) => return Ordering::Less,
    // ...
}
```

Reply:
The case where the polarities were the same would fall through via `=> {}`.